graphene_django/fields.py

```diff
@@ -154,18 +160,61 @@ def resolve_connection(cls, connection, args, iterable, max_limit=None):
         ):
             args["first"] = max_limit
 
+        if graphene_settings.USE_DATALOADERS:
+            if not hasattr(info.context, "dataloaders"):
+                info.context.dataloaders = {}
+
+            dataloader_key: str = str(info.field_nodes)
+            if dataloader_key not in info.context.dataloaders:
+
+                def load_many(keys):
+                    # keys is a list of tuples of (queryset, start, stop)
+                    qs = keys[0][0].model.objects.none()
+
+                    objects = list(
+                        qs.union(
+                            *(
+                                queryset.annotate(
+                                    _dataloader_queryset_index=Value(index)
+                                )[start:stop]
+                                for index, (queryset, start, stop) in list(
```

Comment:
Why is `list()` needed here?

Reply:
I added it because I had noticed perf improvement with it, but I did a more thorough test, and it's useless